src/main/java/fi/vm/yti/datamodel/api/v2/endpoint/DataModelController.java

```diff
@@ -68,52 +72,58 @@ public Datamodel(JenaService jenaService,
-    @PutMapping(path = "/library", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
-    public void createLibrary(@ValidDatamodel(modelType = ModelType.LIBRARY) @RequestBody DataModelDTO modelDTO) {
-        createModel(modelDTO, ModelType.LIBRARY);
+    @ApiResponse(responseCode = "201", description = "The ID for the newly created model")
+    @PostMapping(path = "/library", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> createLibrary(@ValidDatamodel(modelType = ModelType.LIBRARY) @RequestBody DataModelDTO modelDTO) throws URISyntaxException {
+        var uri = createModel(modelDTO, ModelType.LIBRARY);
+        return ResponseEntity.created(new URI(uri)).build();
     }
 
     @Operation(summary = "Create a new application profile")
     @io.swagger.v3.oas.annotations.parameters.RequestBody(description = "The JSON data for the new application profile")
     @ApiResponse(responseCode = "200", description = "The ID for the newly created model")
-    @PutMapping(path = "/profile", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
-    public void createProfile(@ValidDatamodel(modelType = ModelType.PROFILE) @RequestBody DataModelDTO modelDTO) {
-        createModel(modelDTO, ModelType.PROFILE);
+    @PostMapping(path = "/profile", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
+    public ResponseEntity<String> createProfile(@ValidDatamodel(modelType = ModelType.PROFILE) @RequestBody DataModelDTO modelDTO) throws URISyntaxException {
+        var uri = createModel(modelDTO, ModelType.PROFILE);
+        return ResponseEntity.created(new URI(uri)).build();
     }
 
     @Operation(summary = "Modify library")
     @io.swagger.v3.oas.annotations.parameters.RequestBody(description = "The JSON data for the new model node")
     @ApiResponse(responseCode = "200", description = "The ID for the newly created model")
-    @PostMapping(path = "/library/{prefix}", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
-    public void updateLibrary(@ValidDatamodel(modelType = ModelType.LIBRARY, updateModel = true) @RequestBody DataModelDTO modelDTO,
-                              @PathVariable String prefix) {
+    @PutMapping(path = "/library/{prefix}", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
+    public ResponseEntity<Void> updateLibrary(@ValidDatamodel(modelType = ModelType.LIBRARY, updateModel = true) @RequestBody DataModelDTO modelDTO,
+                                           @PathVariable String prefix) {
         updateModel(modelDTO, prefix);
+        return ResponseEntity.noContent().build();
     }
 
     @Operation(summary = "Modify application profile")
     @io.swagger.v3.oas.annotations.parameters.RequestBody(description = "The JSON data for the new model node")
     @ApiResponse(responseCode = "200", description = "The ID for the newly created model")
```

Comment:
Some status codes need to be updated in the documentation, 200 -> 204 or 201

Reply:
Fixed